File: backend/API.DataLayer/Models/Message.cs
namespace API.DataLayer.Models;

public class Message
{
    [Key]
    public Guid MessageId { get; set; }

    [Required]
    public string Content { get; set; } = "";

    public string Image { get; set; } = "";
    public string File { get; set; } = "";
    public string Voice { get; set; } = "";
    public string Video { get; set; } = "";
    public long FileSize { get; set; } = 0;
    public long ImageSize { get; set; } = 0;
    public long VoiceSize { get; set; } = 0;
    public long VideoSize { get; set; } = 0;

    public MessageStatus MessageStatus { get; set; } = MessageStatus.NotSended;

    public Guid ChatId { get; set; }
    public Chat? Chat { get; set; }

    public Guid SenderId { get; set; }
    public User? Sender { get; set; }

    [Required]
    public DateTime SendedAt { get; set; }
}

public enum MessageStatus
{
    NotSended = 0,
    SendedNotSeened = 1,
    SendedAndSeened = 2
}

File: frontend/types/data/message.ts
import {PartialUser} from "./user";

export interface Message {
    messageId: string;
    content: string;
    image: string;
    file: string;
    voice: string;
    video: string;
    fileSize: number;
    imageSize: number;
    voiceSize: number;
    videoSize: number;
    messageStatus: MessageStatus;
    chatId: string;
    sender: PartialUser;
    sendedAt: string;
}

export enum MessageStatus {
    NotSended = 0,
    SendedNotSeened = 1,
    SendedAndSeened = 2,
}

export type PartialMessage = Partial<Message>;

export interface MessageData {
    content: string;
    file?: File | null;
    image?: File | null;
    video?: File | null;
    voice?: File | null;
    chatId: string;
    messageStatus?: MessageStatus;
    senderUserName?: string;
}
